filament/src/fg/fg/ResourceEntry.h

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2019 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef TNT_FILAMENT_FG_RESOURCEENTRY_H
+#define TNT_FILAMENT_FG_RESOURCEENTRY_H
+
+#include "VirtualResource.h"
+
+#include <stdint.h>
+
+namespace filament {
+
+class FrameGraph;
+
+namespace fg {
+
+struct PassNode;
+
+class ResourceEntryBase : public VirtualResource {
+public:
+    explicit ResourceEntryBase(const char* name, uint16_t id, bool imported) noexcept;
+    ResourceEntryBase(ResourceEntryBase const&) = default;
+    ~ResourceEntryBase() override;
+
+    // constants
+    const char* const name;
+    const uint16_t id;                      // for debugging and graphing
+    bool imported;
```

Comment:
Seems like imported (any maybe others?) can be const and private (and therefore renamed to mImported)